expression/constant.go

```diff
@@ -450,3 +451,66 @@ func (c *Constant) Coercibility() Coercibility {
 	}
 	return c.collationInfo.Coercibility()
 }
+
+// TryUpdateEQFunctionCollate Update collate based on priority #35149
+// Try to unify collate by priority before physical optimization.
+// After unifying collate, the impact of session level on some optimizations can be shielded
+func TryUpdateEQFunctionCollate(eq *ScalarFunction) {
```

Comment:
If this function is not used outside the `expression` package, please use lowercase for the function name.

Reply:
OK